pkg/genericapiserver/options/etcd.go

```diff
@@ -45,13 +45,12 @@ func NewEtcdOptions() *EtcdOptions {
 	}
 }
 
-func (s *EtcdOptions) Validate() []error {
-	allErrors := []error{}
+func (s *EtcdOptions) Validate() error {
```

Comment:
I don't think this should keep the same as API object.
Why it needs to return a slice?